bql/semantic/hooks.go

```diff
@@ -664,6 +671,59 @@ func whereObjectClause() ElementHook {
 	return f
 }
 
+// whereFilterClause returns an element hook that updates the working filter clause and,
+// if the filter clause is complete, populates the filters list.
+func whereFilterClause() ElementHook {
+	var f ElementHook
```

Comment:
I prefer to name it `hook` too, but here I just was following the pattern of the rest of this file on which all `ElementHooks` were named just `f` inside of these hook functions. 

Then, we have three options: 
1) Let it as just `f` as the others;
2) Rename all of them to `hook` so they follow the same pattern (different PR for that);
3) Rename only this newly added `ElementHook` to `hook`, letting it different than the others.

How should I proceed?

Reply:
Go readability guidelines would favor `f`

Reply:
Following the [Common Go Code Reviews](https://github.com/golang/go/wiki/CodeReviewComments) page in "golang/go", there is indeed an encouragement for short variable names in Go, but it is also said there that "the further from its declaration that a name is used, the more descriptive the name must be".

This way, since inside the `WhereObjectClauseHook` the `f` variable is used 89 lines away from its declaration, for example, and for consistency through the file (as there are already other `ElementHooks` called `hook` there), it may be better and I was suggested to standardize using `hook` as the name across `hooks.go`.

Then, following this guideline, I have also opened the PR https://github.com/google/badwolf/pull/148.